pyatlan/client/asset.py

```diff
@@ -742,7 +742,7 @@ def _modify_tags(
         propagate: bool = True,
         remove_propagation_on_delete: bool = True,
         restrict_lineage_propagation: bool = True,
-        restrict_propagation_through_hierarchy: bool = False,
+        restrict_hierarchy_propagation: bool = False,
```

Comment:
Why the name change?

Reply:
I've updated the parameter name in tag methods from `restrict_propagation_through_hierarchy` to `restrict_hierarchy_propagation` to maintain consistency with other parameters (e.g., `restrict_lineage_propagation`).

Although this change results in a backward incompatible change, we can mention it as **breaking change** in release `2.1.3`.

Reply:
I think we should be using the same name as on the tag. Personally I think restrict_lineage_propagation so match the value in AtlanTag, but I wouldn't change it now. And since we already did a release with restrict_propagation_through_hierarchy I think we should keep it the same.

Reply:
Okay, cool!

I'll revert that 👍

Reply:
Also it seems to me if you want to only propagate through the hierarchy then it's unclear to me whether you would set restrict_hierarchy_propagation to True or False. It seems to me that restrict_hierarchy_propagation True could mean don't propagate to the hierarchy. On the other hand I can understand wanting consistency and since the properties on AtlanTag will probably be rarely used by the users, I guess consistency is more important.